crates/polars-lazy/src/physical_plan/streaming/convert_alp.rs

```diff
@@ -83,21 +81,18 @@ pub(crate) fn insert_streaming_nodes(
     // this allows us to split at joins/unions and share a sink
     let root = insert_file_sink(root, lp_arena);
 
-    // We use mutation to communicate when we need to insert a file sink.
+    // We use a bool flag in the stack to communicate when we need to insert a file sink.
```

Comment:
It has been a while. So if I understand it correctly we now can have multiple streaming subtrees that all sink into a `hconcat`, where previous with for instance `union`, the union would be part of the streaming subtree and still produce a single `file_sink` that would be picked up by the default engine?

Reply:
Yes that's right, although there are separate sinks for each input stream that become the inputs of the `hconcat`. It looks like the existing code was intended to work this way, but that behaviour was never actually used as all of the plans that have multiple inputs (join and union) are handled explicitly in the `insert_streaming_nodes` match expression rather than falling through to `process_non_streamable_node`.